airbyte-cdk/python/airbyte_cdk/utils/airbyte_secrets_utils.py

```diff
@@ -3,13 +3,15 @@
 #
 
 import logging
-from typing import Any, List, Mapping
+from typing import TYPE_CHECKING, Any, List, Mapping
 
-from airbyte_cdk.sources import Source
 from airbyte_cdk.utils.mapping_utils import all_key_pairs_dot_notation, get_value_by_dot_notation
 
+if TYPE_CHECKING:
+    from airbyte_cdk.sources import Source
 
-def get_secrets(source: Source, config: Mapping[str, Any], logger: logging.Logger) -> List[Any]:
+
+def get_secrets(source: "Source", config: Mapping[str, Any], logger: logging.Logger) -> List[Any]:
```

Comment:
jc why was this needed?

Reply:
The usage of `Source` here for typing caused a circular dependency (this is used by AirbyteTraceMessage). Could also move this `get_secrets` method out into another file since AirbyteTraceMessage doesn't need this specific method (it uses other things defined in this file)